includes/Modules/Analytics/Advanced_Tracking/Measurement_Events/Measurement_Event.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\Measurement_Event
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events;
+
+use \Exception;
+
+/**
+ * Class for representing a single tracking event that Advanced_Tracking tracks.
+ *
+ * @since n.e.x.t.
+ * @access private
+ * @ignore
+ */
+final class Measurement_Event implements \JsonSerializable {
+
+	/**
+	 * The measurement event's configuration.
+	 *
+	 * @since n.e.x.t.
+	 * @var array
+	 */
+	private $config;
+
+	/**
+	 * Measurement_Event constructor.
+	 *
+	 * @since n.e.x.t.
+	 *
+	 * @param array $config The event's configuration.
+	 * @throws \Exception Thrown when config param is undefined.
+	 */
+	public function __construct( $config ) {
+		$this->validate_config( $config );
+
+		if ( ! array_key_exists( 'metadata', $config ) ) {
+			$config['metadata'] = null;
+		}
+
+		$this->config = $config;
+	}
+
+	/**
+	 * Validates the configuration keys and value types.
+	 *
+	 * @since n.e.x.t.
+	 *
+	 * @param array $config The event's configuration.
+	 * @throws \Exception Thrown when invalid keys or value type.
+	 */
+	private function validate_config( $config ) {
+		$valid_keys = array(
+			'pluginName' => false,
+			'category'   => false,
+			'action'     => false,
+			'selector'   => false,
+			'on'         => false,
+			'metadata'   => false,
+		);
+		foreach ( $config as $key => $value ) {
+			if ( ! array_key_exists( $key, $valid_keys ) ) {
+				throw new \Exception( 'Invalid configuration parameter: ' . $key );
+			}
+			$value_type = gettype( $value );
+			if ( 'string' !== $value_type ) {
```

Comment:
This can be simplified to `! is_string( $value )`. We don't need to include the given type in the error, saying that the type must be a string is more helpful.

Reply:
This issue has been resolved.